Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,67 @@ static void Main(string[] args)
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            int totalPosts = exactPostTime.Length;
+            for (int i = 0; i < totalPosts; i++)
+            {
+                for (int j = i + 1; j < totalPosts; j++)
+                {
+                    if (exactPostTime[i] == exactPostTime[j])
+                    {
+                        if (showPostTime[i] != showPostTime[j])
+                        {
+                            Console.WriteLine("Impossible!");
+                            return "impossible";
+                        }
+                    }
+                }
+            }
+
+            string[] resultantCurrentTime = new string[totalPosts];
+            for (int i = 0; i < totalPosts; i++)
+            {
+                string[] splittedTimeIntoHoursMinSec = exactPostTime[i].Split(":");
+                DateTime tempDateTimeObject = new DateTime(2019, 6, 21, Int32.Parse(splittedTimeIntoHoursMinSec[0]), Int32.Parse(splittedTimeIntoHoursMinSec[1]), Int32.Parse(splittedTimeIntoHoursMinSec[2]));
+
+                if (showPostTime[i].Contains("seconds"))
+                {
+                    resultantCurrentTime[i] = exactPostTime[i];
+                }
+                else if (showPostTime[i].Contains("minutes"))
+                {
+                    string spanAfterPostInMinutes = showPostTime[i].Split(" ")[0];
+                    TimeSpan timeSpanObject = new TimeSpan(0, 0, Int32.Parse(spanAfterPostInMinutes), 0);
+                    resultantCurrentTime[i] = tempDateTimeObject.Add(timeSpanObject).ToString().Split(" ")[1];
+                }
+                else
+                {
+                    string spanAfterPostInHours = showPostTime[i].Split(" ")[0];
+                    TimeSpan timeSpanObject = new TimeSpan(0, Int32.Parse(spanAfterPostInHours), 0, 0);
+                    resultantCurrentTime[i] = tempDateTimeObject.Add(timeSpanObject).ToString().Split(" ")[1];
+                }
+            }
+
+            Array.Sort(resultantCurrentTime);
+            Console.WriteLine(resultantCurrentTime[totalPosts - 1]);
+            /*for (int i = 0; i < resultantCurrentTime.Length; i++)
+                Console.WriteLine("-{0}-", resultantCurrentTime[i]);*/
+            string[] temp = resultantCurrentTime[totalPosts - 1].Split(":");
+            string ans;
+            ans = Convert.ToString(Convert.ToInt32(temp[0]) % 24) + ":" + temp[1] + ":" + temp[2];

```

Comment:
Convert.ToString(Convert.ToInt32(temp[0]) % 24) results in failure of second scenario, as it returns 0 not 00.
Use (Convert.ToInt32(temp[0]) % 24).ToString("00") to return string in proper format